src/lib/OutPort.coffee

```diff
@@ -58,6 +59,26 @@ class OutPort extends BasePort
       continue unless socket
       socket.disconnect()
 
+  sendIP: (type, data, options, socketId) ->
+    ip = new IP type, data, options
+    sockets = @getSockets socketId
+    @checkRequired sockets
+    if @isCaching() and data isnt @cache[socketId]?.data
+      @cache[socketId] = ip
+    for socket in sockets
+      continue unless socket
```

Comment:
I think we need to describe some use cases for cloning. Then it will be easier to provide a practical interface for it.